### DISCUSSION TITLE
disable `noFallthroughSwitchClause` with `// falls through` comment

### DISCUSSION THREAD

**Discussion by Birdie0** [biomejs/biome#2802] [Ideas]
In Eslint [no-fallthrough](https://eslint.org/docs/latest/rules/no-fallthrough) rule does not trigger if case body includes `// falls through` or similar comment and I really would like Biome would do that too as `// biome-ignore lint/suspicious/noFallthroughSwitchClause: falls through` is quite long comment to allow fall through case and less readable than first one.

Here's example of disabling rule in [Eslint Playground](https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgbnVtID0gMTtcbnN3aXRjaChudW0pIHtcbiAgY2FzZSAxOiB7XG4gICAgY29uc29sZS5sb2coJ29uZScpXG4gIH1cbiAgY2FzZSAyOiB7XG4gICAgY29uc29sZS5sb2coJ3R3bycpXG4gIH1cbn1cblxuc3dpdGNoKG51bSkge1xuICBjYXNlIDE6IHtcbiAgICBjb25zb2xlLmxvZygnb25lJylcbiAgICAvLyBmYWxscyB0aHJvdWdoXG4gIH1cbiAgY2FzZSAyOiB7XG4gICAgY29uc29sZS5sb2coJ3R3bycpXG4gIH1cbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==) and [Biome Playground](https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAbgB1AG0AIAA9ACAAMQA7AAoAcwB3AGkAdABjAGgAKABuAHUAbQApACAAewAKACAAIABjAGEAcwBlACAAMQA6ACAAewAKACAAIAAgACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAJwBvAG4AZQAnACkACgAgACAAfQAKACAAIABjAGEAcwBlACAAMgA6ACAAewAKACAAIAAgACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAJwB0AHcAbwAnACkACgAgACAAfQAKAH0ACgAKAHMAdwBpAHQAYwBoACgAbgB1AG0AKQAgAHsACgAgACAAYwBhAHMAZQAgADEAOgAgAHsACgAgACAAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACcAbwBuAGUAJwApAAoAIAAgACAAIAAvAC8AIABiAGkAbwBtAGUALQBpAGcAbgBvAHIAZQAgAGwAaQBuAHQALwBzAHUAcwBwAGkAYwBpAG8AdQBzAC8AbgBvAEYAYQBsAGwAdABoAHIAbwB1AGcAaABTAHcAaQB0AGMAaABDAGwAYQB1AHMAZQA6ACAAZgBhAGwAbABzACAAdABoAHIAbwB1AGcAaAAKACAAIAB9AAoAIAAgAGMAYQBzAGUAIAAyADoAIAB7AAoAIAAgACAAIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKAAnAHQAdwBvACcAKQAKACAAIAB9AAoAfQA%3D) for comparison.

**Comment by Conaclos** [comment]
Fall-through is pretty rare, I think it is ok to require a long comment.

**Comment by SirzBenjie** [comment]
There's something else to consider here.  The fall through should go *at the end* of the case body. I believe biome requires that comment to be before the case statement itself.
This creates some additional, albeit minor, cognitive overhead.

I think leaving the comment as the ignore is fine, perhaps.

**Reply by dyc3** [reply]
This is a really good point. However, as seen in the biome playground posted above, you can place the comment at the bottom of the case and it will work as you suggested. But, we don't _require_ it to be at the end of the case.

That being said, it would be better if people migrating to biome had an easier path to do so. Perhaps we could have a code fix for the rule to specifically turn the `// falls through` comment to a proper biome suppression comment?
